### PR TITLE
Add class and package comments with examples

### PR DESCRIPTION
**Pull request by SmiljanaKnezev**

### CODE REVIEW

**Comment by Ducasse** [comment]
Tx I will have a look.

File: src/Containers-OrderPreservingDictionary/CTOrderPreservingIdentityDictionary.class.st [hunk 1]
```diff
@@ -1,5 +1,21 @@
 "
-An IdentityDictionary version that uses == instead of = for key comparing.
+I am an IdentityDictionary version that uses == instead of = for key comparing. That means I consider two keys to be equal only if they are the same object.
```
Comment (guillep) [inline src/Containers-OrderPreservingDictionary/CTOrderPreservingIdentityDictionary.class.st]: This is kind of redundant. A normal dictionary does equality comparisons (`=`). An identity dictionary by definition uses identity comparisons (`==`).

I'd rephrase it maybe like:

> I'm a dictionary that uses identity comparison (==) on my keys instead of equality.
> That means I consider two keys to be equal only if they are the same object.

File: src/Containers-OrderPreservingDictionary/CTOrderPreservingDictionary.class.st [hunk 2]
```diff
@@ -1,6 +1,5 @@
 "
-By default, when CTOrderPreservingDictionaryionary raises an exception. 
-I'm a special kind of order-preserving dictionary that returns a configurable default value (nil by default) instead of raising an exception. 
+I'm a special kind of order-preserving dictionary that returns a configurable default value (nil by default) instead of raising an error.
```
Comment (guillep) [inline src/Containers-OrderPreservingDictionary/CTOrderPreservingDictionary.class.st]: It would be nice to explain what order-preserving means. Elements are stored in the order in which they were added? What happens if I "update" an entry? is it moved down (up?) in the order or it is left in the same order?

Imagine
```smalltalk
dict := CTOrderPreservingDictionary new. 
dict at: 1 put: 'first'.
dict at: 2 put: 'second'.
dict at: 1 put: 'another first'.
```
In which order is 'another first'? I think the comment should specify it (and put a piece of code as an example :)

File: src/Containers-OrderPreservingDictionary/CTOrderPreservingIdentityStringDictionary.class.st [hunk 3]
```diff
@@ -1,5 +1,20 @@
 "
-An IdentityDictionary version that uses == instead of = for key comparing.
+I am an IdentityDictionary version that uses == instead of = for String key comparing. That means I consider two keys to be equal only if they are the same object. 
+
+I also returns empty strings instead of raising errors when absent keys/values are accessed. See CTOrderPreservingStringDictionary comment for example.
```
Comment (guillep) [inline src/Containers-OrderPreservingDictionary/CTOrderPreservingIdentityStringDictionary.class.st]: What's the difference between this one and a normal `CTOrderPreservingIdentityDictionary`?
Why should I use this one and not the other (and vice-versa)?

I'm not saying this class is bad, but the guy reading the documentation will ask herself that question ;)

File: src/Containers-OrderPreservingDictionary/CTStandardOrderedDictionary.class.st [hunk 4]
```diff
@@ -7,6 +7,23 @@ For example, the printing is using the insertion order
 
 	CTStandardOrderedDictionary orderedDictionaryABC printString 
 	>>> 'a CTStandardOrderedDictionary(#A->''A'' #B->''B'' #C->''C'')'
+
+I work mainly as a Dictionary except that I also store keys in an Array to keep track of the order of elements.
```
Comment (guillep) [inline src/Containers-OrderPreservingDictionary/CTStandardOrderedDictionary.class.st:20]: What's the difference between this one and `CTOrderPreservingDictionary`?

File: src/Containers-OrderPreservingDictionary/CTStandardOrderedDictionary.class.st [hunk 5]
```diff
@@ -7,6 +7,23 @@ For example, the printing is using the insertion order
...
+	
+If I try to access key that doesn't exist an error is raised.
```
Comment (guillep) [inline src/Containers-OrderPreservingDictionary/CTStandardOrderedDictionary.class.st]: Maybe the difference is that one has a default value and this one raises an error?
Wouldn't it be good to make that a strategy? (like, if we set a default value we use it, or we can say `beStrict` to raise an error)

File: src/Containers-OrderPreservingDictionary/CTStandardOrderedIdentityDictionary.class.st [hunk 6]
```diff
@@ -1,5 +1,10 @@
 "
-An IdentityDictionary version that uses == instead of = for key comparing.
+I'm an IdentityDictionary version that uses == instead of = for key comparing.That means I consider two keys to be equal only if they are the same object. 
+
+If I try to access key that doesn't exist an error is raised. See CTStandardOrderedDictionary comment for an example.
```
Comment (guillep) [inline src/Containers-OrderPreservingDictionary/CTStandardOrderedIdentityDictionary.class.st]: Same comment as above.

**Review by guillep — CHANGES_REQUESTED** [review]
I've made some question and comments. I don't know the original history of the project, so maybe @Ducasse can give some insights :).

File: src/Containers-OrderPreservingDictionary/CTStandardOrderedIdentityDictionary.class.st (hunk 6 above)
Comment (Ducasse) [reply src/Containers-OrderPreservingDictionary/CTStandardOrderedIdentityDictionary.class.st]: Yes I like the idea of having a strategy. I will integrate this issue and we can always continue

File: src/Containers-OrderPreservingDictionary/CTOrderPreservingIdentityStringDictionary.class.st (hunk 3 above)
Comment (Ducasse) [reply src/Containers-OrderPreservingDictionary/CTOrderPreservingIdentityStringDictionary.class.st]: Totally agree.

**Comment by Ducasse** [comment]
Smiljana did you take into account our remarks.

**Comment by SmiljanaKnezev** [comment]
I have, but was a bit confused should I fix comments if some classes are going to be merged. Guille said that we can merge comments as well. I've been further comparing implementations and decided to set this aside a bit. I collected the differences in a blog post. The plan is to fix the comments now and start discussing how to implement strategy.

File: src/Containers-OrderPreservingDictionary/CTStandardOrderedDictionary.class.st (hunk 4 above)
Comment (SmiljanaKnezev) [reply src/Containers-OrderPreservingDictionary/CTStandardOrderedDictionary.class.st:20]: Only that it is raising an error
